app/kuma-dp/pkg/dataplane/probes/grpc.go

```diff
@@ -0,0 +1,87 @@
+package probes
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials/insecure"
+	grpchealth "google.golang.org/grpc/health/grpc_health_v1"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+
+	"github.com/kumahq/kuma/pkg/version"
+)
+
+func (p *Prober) probeGRPC(writer http.ResponseWriter, req *http.Request) {
+	// /grpc/<port>
+
+	opts := []grpc.DialOption{
+		grpc.WithUserAgent(fmt.Sprintf("kube-probe/%s", version.Build.Version)),
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithContextDialer(func(ctx context.Context, addr string) (net.Conn, error) {
```

Comment:
Should we add timeout?

Reply:
`grpc.WithTimeout` is now deprecated and the recommended way is to attach timeout in the context object when invoking the RPC. We used timeout settings in `context` underneath.